core/embed/trezorhal/stm32u5/secret.c

```diff
@@ -265,7 +288,7 @@ void secret_prepare_fw(secbool allow_run_with_secret, secbool trust_all) {
 #ifdef USE_OPTIGA
   secret_optiga_uncache();
   if (sectrue == allow_run_with_secret) {
-    if (secfalse != secret_optiga_present()) {
+    if (sectrue != secret_optiga_writable()) {
```

Comment:
this flow is rather convoluted, which is probably why we made this mistake

how about flattening it like so:
```
#ifdef USE_OPTIGA
  secret_optiga_uncache();
  secbool optiga_secret_present = secret_optiga_present();
  secbool optiga_secret_writable = secret_optiga_writable();
  if (sectrue == allow_run_with_secret && sectrue == optiga_secret_writable && secfalse == optiga_secret_present) {
    // Secret is not present and the secret sector is writable.
    // This means the U5 chip is unprovisioned.
    // Allow trusted firmware (prodtest presumably) to access the secret sector,
    // early return here.
    return;
  }
  if (sectrue == allow_run_with_secret && sectrue == optiga_secret_present) {
    // Firmware is trusted and the Optiga secret is present, make it available.
    secret_optiga_cache();
  }
  // Disable access unconditionally.
  secret_disable_access();
  if (sectrue != trust_all && sectrue == optiga_secret_present) {
    // Untrusted firmware, locked bootloader. Show the restricted screen.
    show_install_restricted_screen();
  }
#else
  secret_disable_access();
#endif
```

Reply:
Agree that it improves readability. 

I added `sectrue == trust_all` as it matches the previous behavior and i think more restrictions here are good.

https://github.com/trezor/trezor-firmware/pull/4157/commits/85bc0438251f9e0d6f8b333fae8d399afb14f57c